testapi.pm

```diff
@@ -536,7 +541,14 @@ returns the base URL to contact the local os-autoinst service
 
 sub autoinst_url() {
     # move to backend?
-    return "http://10.0.2.2:" . (get_var("QEMUPORT") + 1);
+    if (get_var("CONNECT_PASSWORD")) {    #CONNECT_PASSWORD is used in HACLUSTER
+        use Net::Domain;
+        my $fqdn = Net::Domain::hostfqdn();
```

Comment:
I think this code doesn't talk to openqa but to os-autoinst. 10.0.2.2 is correct there to reach the host from within kvm with user mode networking. It should just be moved to the backend as it's very qemu user mode specific. when using tap devices os-autoinst should probably use the ip address(es) it has on the tap interface where qemu gets connected.
Use of any fqdn guesses is most likely broken by design. don't do that please.